File: python/src/taskulus/issue_close.py
"""Issue close workflow."""

from __future__ import annotations

from pathlib import Path

from taskulus.issue_update import IssueUpdateError, update_issue
from taskulus.models import IssueData


class IssueCloseError(RuntimeError):
    """Raised when issue closing fails."""


def close_issue(root: Path, identifier: str) -> IssueData:
    """Close an issue by transitioning it to closed status.

    :param root: Repository root path.
    :type root: Path
    :param identifier: Issue identifier.
    :type identifier: str
    :return: Updated issue data.
    :rtype: IssueData
    :raises IssueCloseError: If closing fails.
    """
    try:
        return update_issue(
            root=root,
            identifier=identifier,
            title=None,
            description=None,
            status="closed",
        )
    except IssueUpdateError as error:
        raise IssueCloseError(str(error)) from error

File: rust/src/issue_close.rs
//! Issue close workflow.

use std::path::Path;

use crate::error::KanbusError;
use crate::issue_update::update_issue;
use crate::models::IssueData;

/// Close an issue by transitioning it to closed status.
///
/// # Arguments
/// * `root` - Repository root path.
/// * `identifier` - Issue identifier.
///
/// # Errors
/// Returns `KanbusError` if closing fails.
pub fn close_issue(root: &Path, identifier: &str) -> Result<IssueData, KanbusError> {
    update_issue(
        root,
        identifier,
        None,
        None,
        Some("closed"),
        None,
        false,
        true,
        &[],
        &[],
        None,
    )
}
